### ISSUE TITLE
FormatByKey and FormatByKind not work

### ISSUE THREAD

**Issue by KScaesar** [samber/slog-formatter#4]
- FormatByKey Example:
key=time
expected=`2024-09-09T15:01:21+08:00`
actual= `  2024-09-09T15:01:21.505765+08:00`

- FormatByKind Example:
key=latency
expected=1m0s
actual=60000000000

```go
stdHandler := slog.NewJSONHandler(os.Stdout, nil)

formatterHandler := slogformatter.NewFormatterHandler(
	slogformatter.FormatByKey(slog.TimeKey, func(value slog.Value) slog.Value {
		t := value.Time()
		return slog.StringValue(t.Format(time.RFC3339))
	}),
	slogformatter.FormatByKind(slog.KindDuration, func(value slog.Value) slog.Value {
		return slog.StringValue(value.Duration().String())
	}),
)(stdHandler)

t1, _ := time.Parse(time.RFC3339, "2023-08-19T20:00:00+08:00")
t2, _ := time.Parse(time.RFC3339, "2023-08-19T20:01:00+08:00")
var latency time.Duration = t2.Sub(t1)
fmt.Println("go fmt", latency.String())

logger := slog.New(formatterHandler)
logger.Info("format not work",
	slog.Group("response",
		slog.Duration("latency", latency),
	),
)

// 	go fmt 1m0s
// {
//  "time":"2024-09-09T15:01:21.505765+08:00",
//  "level":"INFO",
//  "msg":"format not work",
//  "response":{
//    "latency":60000000000
//  }
// }

```

[playground](https://go.dev/play/p/SfHqVEDuxzQ)

**Comment by samber** [comment]
The `time` attribute is added by the `slog.JSONHandler`, after the processing of `FormatByKey`, so it won't be formatted in  `time.RFC3339`.

You should check the `ReplaceAttrs` argument passed to `NewJSONHandler`.

About the duration, I need to investigate.

**Comment by samber** [comment]
Fixed in v1.1.0.

It was because of the use of `slog.Group("response", ...)`. The FormatByType+FormatByKind+FormatByKey helpers, now navigate through nested attributes.

Thanks for the bug report.
